pilot/pkg/security/authz/model/matcher/metadata.go

```diff
@@ -37,12 +37,12 @@ func MetadataStringMatcher(filter, key string, m *envoy_matcher.StringMatcher) *
 }
 
 // MetadataListMatcher generates a metadata list matcher for the given path keys and value.
-func MetadataListMatcher(filter string, keys []string, value string) *envoy_matcher.MetadataMatcher {
+func MetadataListMatcher(filter string, keys []string, value string, v1beta1 bool) *envoy_matcher.MetadataMatcher {
```

Comment:
nit: we should have unit tests for these public methods. It's easier to check for new changes later.

Reply:
we have `metadata_list.go` but I think it's not covered there for now, but this function is still covered in many other tests, e.g. the `integration_test.go`

Reply:
Added